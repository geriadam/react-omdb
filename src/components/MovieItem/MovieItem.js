import React from 'react';
import styles from './styles.module.css';
import noImage from '../../assets/noImage.png';

const MovieItem = ({ movie, handleClick}) => {
  const poster = movie.Poster === 'N/A' ? noImage : movie.Poster;
  const imgAltText = movie.Poster === 'N/A' ? `Missing poster for ${movie.Title}` : `Poster for ${movie.Title}`;

  const onClickItem = () => {
    handleClick(movie.imdbID);
  }

  return (
    <div className="mx-auto col-sm-6 col-md-4 col-lg-4 mb-5">
      <div className={`card card-body bg-dark text-center h-100 ${styles.card}`} onClick={onClickItem}>
        <img className="w-100 mb-2" src={poster} alt={imgAltText} />
        <h5 className="text-light card-title">
          {movie.Title} - {movie.Year}
        </h5>
      </div>
    </div>
  )
}

export default MovieItem;
