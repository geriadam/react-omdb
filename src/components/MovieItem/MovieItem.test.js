import MovieItem from './MovieItem';

describe('MovieItem', () => {
  describe('poster exists', () => {
    const movie = { Title: 'star wars', Poster: 'star wars poster', Year: '1999', imdbRating: '5/5' };
    let returnValue;
    beforeEach(() => {
      returnValue = MovieItem({ movie });
    });
    it('should create the MovieItem component', () => {
      expect(returnValue).toMatchSnapshot();
    });
  });
  describe('poster does not exist', () => {
    const movie = { Title: 'star wars', Poster: 'N/A', Year: '1999', imdbRating: '5/5' };
    let returnValue;
    beforeEach(() => {
      returnValue = MovieItem({ movie });
    });
    it('should create the MovieItem component', () => {
      expect(returnValue).toMatchSnapshot();
    });
  });
});
