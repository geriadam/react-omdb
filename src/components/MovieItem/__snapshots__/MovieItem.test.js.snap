// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieItem poster does not exist should create the MovieItem component 1`] = `
<div
  className="mx-auto col-sm-6 col-md-4 col-lg-4 mb-5"
>
  <div
    className="card card-body bg-dark text-center h-100 card"
    onClick={[Function]}
  >
    <img
      alt="Missing poster for star wars"
      className="w-100 mb-2"
      src="noImage.png"
    />
    <h5
      className="text-light card-title"
    >
      star wars
       - 
      1999
    </h5>
  </div>
</div>
`;

exports[`MovieItem poster exists should create the MovieItem component 1`] = `
<div
  className="mx-auto col-sm-6 col-md-4 col-lg-4 mb-5"
>
  <div
    className="card card-body bg-dark text-center h-100 card"
    onClick={[Function]}
  >
    <img
      alt="Poster for star wars"
      className="w-100 mb-2"
      src="star wars poster"
    />
    <h5
      className="text-light card-title"
    >
      star wars
       - 
      1999
    </h5>
  </div>
</div>
`;
